volatility/framework/layers/registry.py

```diff
@@ -227,8 +227,14 @@ def dependencies(self) -> List[str]:
 
     def is_valid(self, offset: int, length: int = 1) -> bool:
         """Returns a boolean based on whether the offset is valid or not."""
-        # TODO: Fix me
-        return True
+        try:
+            # Pass this to the lower layers for now
+            return all([
+                self.context.layers[layer].is_valid(offset, length)
+                for (_, offset, length, layer) in self.mapping(offset, length)
+            ])
+        except exceptions.InvalidAddressException:
```

Comment:
Is there a reason this one catches an exception still?

Reply:
Yep, the mapping call itself may still throw an `InvalidAddressException`.  We've also got a case where `is_valid` can sometimes call `mapping` so we don't often let `mapping` use `is_valid`, so I think this catch still needs to be there.